[Tooling] Fix .po file to send new Jetpack PlayStore Listing copies to translation

AliSoftware opened the pull request:
# Why

Recently, the copies for the Jetpack Play Store Listing were updated by @zwarm via https://github.com/wordpress-mobile/WordPress-Android/pull/16924/, but unfortunately, those new copies were never included in the `.po` file and thus sent to translation.

After investigation, it turns out that the `.po` file uses the `.txt` files from `WordPress/jetpack_metadata/*.txt` as the source of truth to generate the `PlayStoreStrings.po` file, but the files that were updated in https://github.com/wordpress-mobile/WordPress-Android/pull/16924/ were the ones in `fastlane/jetpack_metadata/android/en-US/*.txt` instead.

# How

### First attempt: renaming the keys

At first I thought the reason why the copies had not been included in the `.po` file was because the keys used in the `Fastfile` and the ones used in the `.po` file mismatched, and since we have a bug in our action generating the `.po` file that makes it only be able to update entries in the `.po` file if those keys already exist (see pdnsEh-jX-p2) that would have explained why the `.po` file failed to be updated.
This is why my first commits change the keys used in the `fastlane/lanes/localization.rb` and `PlayStoreStrings.po` files.

After realizing that this didn't solve the problem, I still decided to keep those new keys, because that would make them consistent with the keys we use for WordPress, and more in line with the Android vocabulary.

### Second attempt: updating the real source of truth

After realizing that the real issue was that the `.txt` files that had been updated were not the expected one that are used as source of truth for the `.po` file, I did a `cp fastlane/jetpack_metadata/android/en-US/*.txt WordPress/jetpack_metadata/` to copy the files that were changed in https://github.com/wordpress-mobile/WordPress-Android/pull/16924 back in the expected location.

Then I ran `bundle exec fastlane update_appstore_strings` on top of that, and finally the `.po` file got updated as expected.

# Targeting the release branch

Because, [during `finalize_release`, we will end up calling `download_metadata_strings`](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/fastlane/lanes/release.rb#L145-L156), which amongst other things, [copies the files from `WordPress/jetpack_metadata/*.txt` into `fastlane/jetpack_metadata/android/en-US` (since English is the original for GlotPress and is thus not downloaded from it)](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/fastlane/lanes/localization.rb#L265-L272), I needed to make that fix in the `release/20.4` branch, and land it before finalizing the release, otherwise during the next `finalize_release` tomorrow, the changes from https://github.com/wordpress-mobile/WordPress-Android/pull/16924 would be overwritten by the old `WordPress/jetpack_metadata/*.txt` copies.

# What's Next

The next time the `release/20.4` branch will land to `trunk`, the wpcom cron job should then re-import the updated `PlayStoreStrings.po` file from `trunk` into the Jetpack GlotPress project, and those will then be sent to our translation vendor and start being translated during next sprint.

AliSoftware committed:
Update keys in Jetpack's PlayStoreStrings.po

AliSoftware committed:
Fix code typo

AliSoftware committed:
Copy new originals in the right place

When @zwarm updated the copies in https://github.com/wordpress-mobile/WordPress-Android/pull/16924, they updated the `fastlane/jetpack_metadata/android/en-US/` files while the source of truth to update the `.po` file before translations is in `WordPress/jetpack_metadata/`

I agree that this is confusing and makes it hard to know which ones is the source of truth, especially since for the specific case of `en-US` (which is the source locale so not downloaded back from GlotPress), we end up just doing a copy of the file from `WordPress/jetpack_metadata/` into `fastlane/jetpack_metadata/android` (see https://github.com/wordpress-mobile/WordPress-Android/blob/713cf754e168764d7025b2ddbcf63fc82739b27e/fastlane/lanes/localization.rb\#L265-L272)

Hopefully we will find some time in the future to clean that up and make that a bit less confusing, having only a single and clear source of truth instead of people wondering if that source of truth is `WordPress/jetpack_metadata` or `fastlane/jetpack_metadata/default` or something else… One day…

AliSoftware committed:
Update Jetpack `PlayStoreStrings.po` for version 20.4

ParaskP7 commented on fastlane/lanes/localization.rb:
```diff
@@ -156,13 +156,13 @@
-      'app-store-name': File.join(metadata_folder, 'title.txt'),
+      play_store_promo: File.join(metadata_folder, 'short_description.txt'),
+      play_store_desc: File.join(metadata_folder, 'full_description.txt'),
+      play_store_app_title: File.join(metadata_folder, 'title.txt'),
```
Suggestion (💡): Consider ordering those 3, `app_title`, `promo` and `desc` in that order, just like it is done elsewhere. And actually, even if at some places it is not ordered as such (ie. line `125`), consider reordering those too, just so that they end up being consistent.
- `play_store_app_title`
- `play_store_promo`
- `play_store_desc`

AliSoftware replied:
Good idea! Addressed in 3c1251d66034b311d7b045330ab95ff9b32bb2dc

ParaskP7 replied:
Awesome, thanks for updating the `localization.rb` as well! 🤟

AliSoftware replied:
To be honest that odd order in the `.po` file was bothering me for some time as well 😛 So it was high time to make it more consistent 😄

ParaskP7 commented on fastlane/lanes/localization.rb:
```diff
@@ -156,13 +156,13 @@
     }
 
     update_po_file_for_metadata_localization(
-      po_path: metadata_folder = File.join(metadata_folder, 'PlayStoreStrings.po'),
```
Praise (❤️): Good catch!

ParaskP7 reviewed (approved):
👋 @AliSoftware !

I reviewed this PR and everything LGTM, it is a 👍 from my side! 💯 ✅

I disabled the auto-merge because:
- I want to give the chance to @zwarm and/or @enejb to quickly scan the changes as well, especially the `title.txt`, `short_description.txt` and `full_description.txt` changes (although you just copy-pasted them Olivier from the `en-US` folder).
- I want to give you Olivier the chance to apply this single suggestion (💡) of mine, just in case you might want to.

AliSoftware committed:
Reorder PlayStoreStrings keys for consistency

h/t @ParaskP7 for the suggestion in https://github.com/wordpress-mobile/WordPress-Android/pull/17005#discussion_r938635376

zwarm reviewed (approved):
@AliSoftware - The copy looks identical, thank you for copying it over. 

❓  If the source of truth is `WordPress/jetpack_metadata/*.txt`, what is the purpose of ` fastlane/jetpack_metadata/android/en-US/*.txt` . I see where they get updated, but do we need both places?

UPDATE: I think I answered my own question - that is en-US. Easy mistake to make.

AliSoftware commented:
> @AliSoftware - The copy looks identical, thank you for copying it over.

Thanks for the check!

> ❓ If the source of truth is `WordPress/jetpack_metadata/*.txt`, what is the purpose of ` fastlane/jetpack_metadata/android/en-US/*.txt` . I see where they get updated, but do we need both places?
> 
> UPDATE: I think I answered my own question - that is en-US. Easy mistake to make.

Yeah it's a bit confusing I agree.
 - `WordPress/{jetpack_}metadata/*.txt` are the source of truth that will then be inserted into the `PlayStoreStrings.po` file to be sent to translation
 - `fastlane/{jetpack_}metadata/android/$locale/*.txt` are the translations then downloaded back from GlotPress.
    - Because, to GlotPress, `en-US` is not a "translation" but the "original", that means that when we ask GlotPress to export all the translations for all the locales it got, it exports everything except `en-US`, which it doesn't consider a translation. Which is why for this specific case of `en-US`, we copy over the originals (the source of truth we used from `WordPress/{jetpack_}metadata/*.txt` into `fastlane/{jetpack_}metadata/android/en-US/`) ourselves.
    - Those files are then used by `fastlane` itself (more precisely, by its tool [`supply`](https://docs.fastlane.tools/actions/supply/)) which takes those `.txt` files and upload their content to the PlayStore Main Listing, via the PlayStore Console API. Because those paths are what fastlane uses by convention for those, we need to keep those organized the way they are (`$locale/*.txt` files, with the name of the txt files obeying fastlane's convention).
    - And that's also the reason why we need to copy the original copies from `WordPress/{jetpack_}metadata/*.txt` into `fastlane/{jetpack_}metadata/android/en-US`, so that fastlane sees those copies in the folder structure it expects and thus also updates the metadata for `en-US` when it updates them using the PlayStore API.

_PS: In the future I hope to make the `fastlane/{jetpack_}metadata/android/en-US/` folder be the source of truth for our `PlayStoreStrings.po` file and get rid of `WordPress/{jetpack_}metadata/*.txt` for those, to avoid confusion. Though there's still the case of release notes, for which we might need to have separate source of truth between "what's extracted at code-freeze time but is not yet editorialized" and "what's editorialized and ready to go for translation and to be used for PlayStore"…_

AliSoftware commented:
CI failure on the `instrumented-tests` is a known failure — `wPScreenshotTest` failing, they have been disabled on `trunk` to workaround while we work on fixing that, but haven't been disabled on `release/20.4`.

I'm thus merging the PR despite that one failing, as we agreed that this failure can be ignored.
